doc/cascadia/Cascading-Default-Settings.md

```diff
@@ -0,0 +1,399 @@
+  matching a default profile loads the user settings into the object created
+  from the defaults.
+* We'll need to make sure that there's only one action bound to each key chord
+  for a keybinding. If there are any key chords in the user settings that match
+  a default key chord, we should bind them to the action from the user settings
+  instead.
+* For any color schemes who's name matches the name of a default color scheme,
+  we'll need to apply the user settings to the existing color scheme.
+
+### Hiding Default Profiles
+
+What if a user doesn't want to see one of the profiles that we've included in
+the default profiles?
+
+We will add a `hidden` key to each profile, which defaults to false. When we
+want to mark a profile as hidden, we'd just set that value to `true`, instead of
+trying to look up the profile's guid.
+
+
+So, if someone wanted to hide the default cmd.exe profile, all they'd have to do
+is add `"hidden": true` to the cmd.exe entry in their user settings, like so:
+
+```js
+{
+    "profiles": [
+        {
+            // Make changes here to the cmd.exe profile
+            "guid": "{6239a42c-1de4-49a3-80bd-e8fdd045185c}",
+            "hidden": true
+        }
+    ],
+```
+
+#### Hidden Profiles and the Open New Tab shortcuts
+
+Currently, there are keyboard shortcuts for "Open New Tab With Profile
+&lt;N&gt;". These shortcuts will open up the Nth profile in the new tab
+dropdown. Considering we're adding the ability to remove profiles from that
+list, but keep them in the overall list of profiles, we'll need to make sure
+that the handler for that event still opens the Nth _visible_ profile.
+
+### Dynamic Profiles
+
+Sometimes, we may want to auto-generate a profile on the user's behalf. Consider
+the case of WSL distros on their machine, or VMs running in Azure they may want
+to auto-connect to. These _dynamic_ profiles have a source that might be added
+or removed after the app is installed, and they will be different from user to
+user.
+
+Currently, these profiles are only generated when a user first launches the
+Terminal. If they already have a `profiles.json` file, then we won't run the
+auto-generation behavior. This is obviously not great - if any new types of
+dynamic profiles are added, then users that already have the Terminal installed
+won't get any of these dynamic profiles. Furthemore, if any of the sources of
+these dynamic profiles are removed, then the app won't auto-remove the
+associated profile.
+
+In the new model, with a combined defaults & user settings, how should these
+dynamic profiles work?
+
+I propose we add functionality to automatically search for these profile sources
+and add/remove them on _every_ Terminal launch. To make this functionality work
+appropriately, we'll need to introduce a constraint on dynamic profiles.
+
+**For any dynamic profiles, they must be able to be generated using a stable
+GUID**. For example, any time we try adding the "Ubuntu" profile, we must be

```

Comment:
ooh ooh with our `generate uuid v5` function we can request that every dynamic profile provider have its own namespace GUID, so that..
if the user has an Azure VM named `cmd`, it won't guid-conflict with the CascadiaDefaults `cmd` guid

Reply:
in fact, we can _force_ the provider to tell us its _namespace_ guid at runtime

Reply:
See I was just gonna rely on the provider figuring out how to make its own static GUIDs. Presumably each Azure VM has a static GUID, right? I figured they'd use that.

Reply:
possibly.. i guess WSL profiles have GUIDs too, but.. those are hidden from us!

Reply:
You could theoretically.... make a namespace to feed into `generate uuid v5` by stitching whatever string the provider gives us to the end of a reasonable namespace.

ExtensionName.ProviderIdentifier
"Microsoft.AzureCloudShell" + "{VMGUID}" = "Microsoft.AzureCloudShell.{VMGUID}" --> uuid v5 --> {Stable GUID}
"Fred.Extension" + "Cookies" = "Fred.Extension.Cookies" --> uuid v5 --> {Stable GUID}

Then the provider CAN give you a GUID. Or it CAN give you some other stable string.

Reply:
So I guess the point I'm struggling with here is how we get the provider to communicate a stable ID to us, if they don't generate it themselves.

So lets examine the WSL case. Lets say the generator namespace is `Microsoft.Terminal.WSL`. We can always turn `Microsoft.Terminal.WSL` into a uuidv5 and stamp that into the `source` of any profiles it gives us. However, we still need them to give us unique, stable guids for the profiles, or otherwise somehow communicate a stable name to us. 

I guess we could expose some sort of function for the generators to request we make a uuiv5 for a name on their behalf, as long as they know that the names they're using are unique. 

So we'll be able to create a namespace guid for each generator, and the generator would be able to ask us to make guids in that namespace for them.

okay yea I think I understand this now.